features/netsocket/NetworkStack.cpp

```diff
@@ -62,6 +62,36 @@ nsapi_error_t NetworkStack::gethostbyname(const char *name, SocketAddress *addre
     return nsapi_dns_query(this, name, address, interface_name, version);
 }
 
+nsapi_error_t NetworkStack::getaddrinfo(const char *hostname, SocketAddress *hints, SocketAddress **res, const char *interface_name)
+{
+    int i;
+    if (hostname[0] == '\0') {
+        return NSAPI_ERROR_PARAMETER;
+    }
+    nsapi_version_t version = hints->get_ip_version();
+    // if the version is unspecified, try to guess the version from the
+    // ip address of the underlying stack
+    if (version == NSAPI_UNSPEC) {
+        SocketAddress testaddress;
+        if (testaddress.set_ip_address(this->get_ip_address())) {
+            version = testaddress.get_ip_version();
+        }
+    }
+
+    SocketAddress *temp = new (std::nothrow) SocketAddress [MBED_CONF_NSAPI_DNS_ADDRESSES_LIMIT];
```

Comment:
`temp` can now be NULL in case of out-of-memory, case, so should there be check for its validity? `nsapi_dns_query_multiple` does not seem to check it either.

Reply:
Done